### ISSUE TITLE
About Scene Router and the contrastive loss

### ISSUE THREAD

**Issue by cocowy1** [chzhang18/RAG#1]
Hi, Cheng Hao,
Congratulations! This paper made a huge contribution to the continuous learning of stereo matching and inspired me a lot.
Recently, I've been trying to reproduce your work, and I've run into some trackable issues.
1) The scene-router module doesn't seem to exist in your code? I searched carefully, but found nothing. So, could you point out where the problem is?
2) Where is the contrastive loss? It seems that the loss function involved L1 smooth loss.
Sorry to bother you, but to be honest, you push forward the development of stereo matching community!

**Comment by chzhang18** [comment]
@cocowy1 Hi, Thank you for your interest in our work!
- A1: The proposed _Scene Router_ module is not included in the released code. The module is inspired from Expert Gate. To reproduce the module, you can refer to the following code link: [https://github.com/wannabeOG/ExpertNet-Pytorch](ExpertNet).
- A2: As described in Sec. 4.3 of the paper, the proposed _Scene Contrastive Loss_ is to make the reconstructed output close to the input while far from the reconstructed output obtained by the autoencoder trained on the old task. Notably, the Scene Contrastive Loss do not involve the smooth L1 loss.
